tests/test_parsley.py

```
@@ -121,24 +121,43 @@ def test_parse_sensor_analog(self):
         assert res == expected_res
         
+    # ── Error-path shape invariant ──────────────────────────────────────
+    # All `parsley.parse()` error returns MUST carry the same triage fields
+    # as a success return. Previously these tests only checked
+    # `'error' in res['data']`, which let bug #B2 hide: the error dict was
+    # silently dropping `msg_prio` and `msg_metadata` for the entire
+    # 2026.3 cycle and no test failed. The shape check below catches that.
+    _EXPECTED_ERROR_KEYS = {
+        'msg_prio', 'board_type_id', 'board_inst_id',
+        'msg_type', 'msg_metadata', 'data'
+    }
```

Ruff (0.15.13)

[warning] Mutable default value for class attribute

(RUF012)